ostd/src/mm/page_table/cursor.rs

```diff
@@ -225,7 +224,7 @@ where
 
             match self.cur_entry().to_owned() {
```

Comment:
The internal function `cur_entry` gets an `EntryMut`, is there a scenario where `cur_entry` does not require mut ( it seems like here)? I think an additional `cur_entry_mut` could be provided, or directly offer `cur_node` and use `entry`/`entry_mut` following it.